### PR TITLE
Refactor flash message for browse submissions to use partial, closes #2258

### LINKED ISSUE #2258: Refactor flash message when browsing submissions
Refactor the code used to generate flash messages on the browse submissions page to use the general `ApplicationHelper` for setting the flash message, and the `shared/flash_message` partial for rendering it

### PR DESCRIPTION
**Pull request by jeremygoh**
Switch to use `ApplicationHelper` to set our flash notice message. 

Moved logic for determining the notice message into the controller, use shared partial for flash messages to render.

**Test Plan**
Manual testing. Verified notice messages appropriate to the assignment's status appeared.

### CODE REVIEW

File: app/controllers/submissions_controller.rb [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 class SubmissionsController < ApplicationController
   include SubmissionsHelper
+  include PaginationHelper
```
Comment (david-yz-liu) [inline app/controllers/submissions_controller.rb]: Does this really need to be included?

Comment (jeremygoh) [inline app/controllers/submissions_controller.rb]: Thanks for the catch. I must have missed it or made a mistake during a rebase.

File: app/controllers/submissions_controller.rb [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 class SubmissionsController < ApplicationController
   include SubmissionsHelper
+  include ApplicationHelper
```
Comment (david-yz-liu) [inline app/controllers/submissions_controller.rb]: @jeremygoh Actually, can you try without the `ApplicationHelper` as well? I seem to recall it's already included in ApplicationController.

Comment (jeremygoh) [inline app/controllers/submissions_controller.rb]: @david-yz-liu - thanks! Removed the duplicate

**Pull request merged**